bee-message/src/payload/mod.rs

```diff
@@ -0,0 +1,325 @@
+// Copyright 2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+//! The payload module defines the core data types for representing message payloads.
+
+pub mod data;
+pub mod drng;
+pub mod fpc;
+pub mod indexation;
+pub mod salt_declaration;
+pub mod transaction;
+
+use crate::{MessagePackError, MessageUnpackError, ValidationError};
+
+use data::{DataPackError, DataPayload, DataUnpackError};
+use drng::{
+    ApplicationMessagePayload, BeaconPayload, CollectiveBeaconPayload, DkgPackError, DkgPayload, DkgUnpackError,
+};
+use fpc::{FpcPackError, FpcPayload, FpcUnpackError};
+use indexation::{IndexationPackError, IndexationPayload, IndexationUnpackError};
+use salt_declaration::{SaltDeclarationPackError, SaltDeclarationPayload, SaltDeclarationUnpackError};
+use transaction::{TransactionPackError, TransactionPayload, TransactionUnpackError};
+
+use bee_packable::{PackError, Packable, Packer, UnpackError, Unpacker};
+
+use alloc::boxed::Box;
+use core::{convert::Infallible, fmt};
+
+/// Maximum length (in bytes) of a message payload.
+pub const PAYLOAD_LENGTH_MAX: usize = 65157;
```

Comment:
Here it might be helpful to mention in a comment how this rather odd looking number comes to be. I know from the network stuff that the MAX_PACKET_LENGTH is 2^16. A link to a spec is also fine...just something to help understanding this number.